pkg/imgpkg/bundle/bundle_images_lock.go

```diff
@@ -15,6 +15,64 @@ import (
 	"github.com/k14s/imgpkg/pkg/imgpkg/lockconfig"
 )
 
+func (o *Bundle) AllImagesLock() (*ImagesLock, error) {
+	imgsLock, err := o.buildAllImagesLock(map[string]bool{})
+	if err != nil {
+		return nil, err
+	}
+
+	imgsLock.AddImageRef(lockconfig.ImageRef{
+		Image: o.DigestRef(),
+	})
+	return imgsLock, nil
+}
+
+func (o *Bundle) buildAllImagesLock(processedImgs map[string]bool) (*ImagesLock, error) {
```

Comment:
if we are not using bool value itself, common convention is to use a struct instead

```
processedImgs map[string]struct{}
```

then use

```
processedImgs[image.Image] = struct{}{}
```